### PR TITLE
[2.1] Enforce consistent pre-initialization behavior

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

`computeCurrentPriceRatio`, `computeCurrentFourthRootPriceRatio` and `computeCurrentVirtualBalances` would revert if called between registration and initialization. Since "current" values of anything don't make sense before the pool is initialized, revert cleanly with the existing PoolNotInitialized error (vs. cryptic numeric / divide by zero).

`computeCurrentPoolCenteredness` and `isPoolWithinTargetRange` don't revert: but return nonsensical / misleading values. Those should also revert, since they don't have meaningful values before the pool is active.

The edge cases are `getReClammPoolDynamicData` and `computeCurrentPriceRange`. These are unchanged. They shouldn't revert (and there are tests ensuring that they don't), because they can return meaningful data pre-initialization (partial meaningful data + zeros in the case of `getReClammPoolDynamicData`).

Reordered the interface and implementation, splitting the "Pool State Getters" into three semantically coherent groups: storage getters, live getters (to which the changes and block comment apply), and off-chain helpers (exclusively called by aggregators/front-ends, etc.) 

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [X] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [X] Complex code has been commented, including external interfaces
- [X] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge (2.1)

### CODE REVIEW

File: contracts/ReClammPool.sol [hunk 1]
```diff
@@ -405,10 +405,28 @@ contract ReClammPool is IReClammPool, BalancerPoolToken, PoolInfo, BasePoolAuthe
...
+    // Two adjacent read paths are intentionally NOT gated this way:
+    //   - `computeCurrentPriceRange` returns the configured `_INITIAL_MIN_PRICE` / `_INITIAL_MAX_PRICE` before init,
+    //     because the configured range is a meaningful answer to "what range will this pool operate in?". The
+    //     ratio/centeredness/within-range concepts have no comparable meaningful pre-init answer.
+    //   - `getReClammPoolDynamicData` doesn't revert; it leaves the live-state-derived fields zeroed out so that
+    //     integrations can still inspect static fields (rates, fees, supply) on a freshly deployed pool.
+    //
+    // Callers who want the configured initial values (e.g. the implied initial price ratio) should read them from
+    // `getReClammPoolImmutableData()` and compute as needed, not call these getters.
+
```
Comment (jubeira) [inline contracts/ReClammPool.sol]: not a strong opinion, but perhaps this comment makes more sense in the interface rather than here?

I'll leave the final decision to you :)

**Review by jubeira — DISMISSED** [review]
LGTM, makes sense!

File: contracts/ReClammPool.sol (hunk 1 above)
Comment (EndymionJkb) [reply contracts/ReClammPool.sol]: I went back and forth on that. It applies to more than one function... but that could also go in the interface just as easily. Yeah, maybe I'll move it.

In fact, it's too general as written anyway. "view functions below" is ambiguous. I'll break them into semantically coherent sections, and put that comment exactly where it applies. Will hose the diff a bit, I'm afraid, but is much clearer for readers.

**Review by joaobrunoah — DISMISSED** [review]
LGTM!

**Review by jubeira — APPROVED** [review]
A little bit over the top for this stage IMO, but LGTM.

I've verified (manually checking each function vs `reclamm-2.1` and with Claude) that the only behavioral change is the introduction of `onlyWhenInitialized` for these getters:
- `computeCurrentVirtualBalances()`
- `computeCurrentPriceRatio()`
- `computeCurrentFourthRootPriceRatio()`
- `computeCurrentPoolCenteredness()`
- `isPoolWithinTargetRange()`

**Review by joaobrunoah — APPROVED** [review]
LGTM! Also checked manually and with Claude.
